### PR TITLE
OHRM5X-408: Leave Type FE

### PR DESCRIPTION
**Pull request by KaveeshaSilva1996**

### CODE REVIEW

File: symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      leaveType: {...leaveTypeModel},
+      rules: {
+        name: [required, shouldNotExceedCharLength(50)],
+        situational: [],
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue:106]: remove `situational` if unused

File: symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            :rules="rules.name"
+            required
+          />
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :cols="2" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <oxd-input-group
+                :classes="{wrapper: '--status-grouped-field'}"
+                label="Is Entitlement Situational"
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue:42]: Need to use translation strings from leave module onward. add your word to `/symfony/client/messages.en_US.json`
```
{
  "source_language": "en_US",
  "target_language": "en_US",
  "date": "2021-07-19T16:21:30Z",
  "modules": {
    ....
    ....
    "leave": {
      "version": "5.0",
      "messages": {
        ....
        ....
        "is_entitlement_situational": {
          "source": "Is entitlement situational",
          "target": "Is entitlement situational",
          "description": "By clicking the “Is Entitlement Situational” option, the leave type will be excluded from reports unless there is some activity. For example: maternity leave, jury duty leave."
        }
      }
    }
  }
}
```

and use it as `:label="$t('leave.is_entitlement_situational')"`

File: symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+              >
+                <oxd-input-field
+                  type="radio"
+                  v-model="leaveType.situational"
+                  optionLabel="Yes"
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/pages/leaveType/EditLeaveType.vue:47]: use translations for `optionLabel`

File: symfony/client/src/orangehrmLeavePlugin/pages/leaveType/SaveLeaveType.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      this.http
+        .create({
+          name: this.leaveType.name,
+          situational: this.leaveType.situational,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .then(() => {
+          this.leaveType.name = '';
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmLeavePlugin/pages/leaveType/SaveLeaveType.vue:123]: you can use model to reset. `this.leaveType = {...leaveTypeModel}`

**Review by Super-Chama — CHANGES_REQUESTED** [review]
Please use translations for all strings and fix other minor issues

**Review by Super-Chama — APPROVED** [review]

File: installer/dbscript/dbscript-2.sql [hunk 5]
```diff
@@ -3564,6 +3564,8 @@ UPDATE `ohrm_menu_item` SET `status` = '0' WHERE `ohrm_menu_item`.`menu_title` =
 UPDATE `ohrm_menu_item` SET `status` = '0' WHERE `ohrm_menu_item`.`menu_title` = 'Reports' AND `screen_id` IS NOT NULL;
 UPDATE `ohrm_menu_item` SET `status` = '0' WHERE `ohrm_menu_item`.`menu_title` = 'Data Import';
 
+UPDATE `ohrm_menu_item` SET `status` = '1' WHERE `ohrm_menu_item`.`menu_title` = 'Leave Types';
```
Comment (RajithaKumara) [inline installer/dbscript/dbscript-2.sql:3567]: remove this line. menu items enable once leave period defined. Hope data group permission already defined for this API.

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/dbscript/dbscript-2.sql (hunk 5 above)
Comment (KaveeshaSilva1996) [reply installer/dbscript/dbscript-2.sql]: removed

**Pull request merged**